### PR TITLE
Adding Clerk.configuration and config for silence_warnings and logger 

### PR DESCRIPTION
**Pull request by anlek**
Hey, I refactored a bunch of stuff relating to the warning message. Hopefully you don't mind.
I've tried to keep everything within a similar coding style.
I tried to make sure test coverage and even added testing for the warning messages that weren't there originally ;)

Let me know if you need anything tweaked.

### CODE REVIEW

**Comment by anlek** [comment]
Hmm, seems like travis-ci failed due to rake gem, this isn't something I messed with though 😞

**Comment by house9** [comment]
Travis is failing on some of the older versions of Ruby, its green on most of the modern ones; Probably going to change the build configuration soon.

File: README.md [hunk 1]
```diff
@@ -86,6 +86,18 @@ Since we are using `sentient_user` you can also set the updater or creator in te
...
+## Configuration
+
+In case you need to customize Clerk. Just create `config/initializers/clerk.rb` and set up your configuration:
+
+```ruby
+Clerk.configure do |config|
+  config.silence_warnings = true | false     # Defaults: false (true in test env)
+  config.logger = MyCustomLogger.new()       # Defaults: Rails.logger
+end
+```
```
Comment (house9) [inline README.md]: 👍

File: lib/clerk/callback.rb [hunk 2]
```diff
@@ -4,21 +4,33 @@ def before_create(record)
...
       User.current
-    end  
-    
+    end
+
+    #######
+    private
+    #######
+
+    def warn(message)
```
Comment (house9) [inline lib/clerk/callback.rb]: 👍

File: spec/clerk/configuration_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+require 'spec_helper'
+
+describe Clerk::Configuration do
+  before(:each) do
+    # Setups configuration
```
Comment (house9) [inline spec/clerk/configuration_spec.rb]: I would remove the comment, I think the code speaks for itself.

File: spec/clerk/configuration_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    Clerk.configure {}
+  end
+  after(:each) do
+    # Reset configuration
```
Comment (house9) [inline spec/clerk/configuration_spec.rb]: I would remove the comment, I think the code speaks for itself.

File: spec/clerk/configuration_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    Clerk.configuration = Clerk::Configuration.new
+  end
```
Comment (house9) [inline spec/clerk/configuration_spec.rb]: code style: add a line break between the `it`, `before` and `after` blocks

**Comment by anlek** [comment]
For future reference, this is to fix issue #5

File: spec/clerk_spec.rb [hunk 6]
```diff
@@ -76,4 +76,68 @@ class FooExtended < Foo
...
+      it "should log a message" do
+        @logger.should_receive(:warn).with("WARNING: User#current is not defined, are you including SentientUser on your User model?")
+        this = FooExtended.new(:bar => "Test")
+        this.save
+      end
+
+      context "amd warning are silenced" do
+        before do
+          Clerk.configure { |c| c.silence_warnings = true }
+        end
```
Comment (house9) [inline spec/clerk_spec.rb]: code style: add newline between blocks

File: spec/clerk_spec.rb [hunk 7]
```diff
@@ -76,4 +76,68 @@ class FooExtended < Foo
...
+
+        it "should not show a log message" do
+          @logger.should_not_receive(:warn).with("WARNING: User#current is nil, are you including SentientController on your ApplicationController?")
+          this = FooExtended.new(:bar => "Test")
+          this.save
+        end
+      end
+    end
```
Comment (house9) [inline spec/clerk_spec.rb]: code style: remove empty line

File: lib/clerk/configuration.rb [hunk 8]
```diff
@@ -0,0 +1,21 @@
+require 'active_support/logger'
```
Comment (house9) [inline lib/clerk/configuration.rb]: code style: add new line between module and require

File: lib/clerk/railtie.rb [hunk 9]
```diff
@@ -5,6 +5,12 @@ module Clerk
   class Railtie < Rails::Railtie
     initializer 'clerk.ar_extensions' do |app|
       ActiveRecord::Base.extend Clerk
+
+      #initialize configuration object
```
Comment (house9) [inline lib/clerk/railtie.rb]: I would remove the comment, I think the code speaks for itself.

File: lib/clerk/railtie.rb [hunk 10]
```diff
@@ -5,6 +5,12 @@ module Clerk
...
+      Clerk.configure do |config|
+        config.silence_warnings = Rails.env.test?
```
Comment (house9) [inline lib/clerk/railtie.rb]: I'm not a 100% sure about defaulting to true for test environment?

Let's go with it for now - might change this in the future; I guess it is easy enough to enable in spec helper

Comment (anlek) [inline lib/clerk/railtie.rb]: Yeah, I was originally not going to default to true in test env, but as I was writing the docs, it make it seem like you needed another step to get it to work how you expect it to work (aka, no warning messages in your tests).

Can you send me the rubocop.yml file and I can fix it based on that (and maybe add it to the project as well 😜)

**Comment by house9** [comment]
Published: https://rubygems.org/gems/clerk/versions/0.2.3

Thanks!

**Comment by anlek** [comment]
Awesome! Thanks for the speedy PR acceptance!
I can now continue my project without all those warnings! ;)

**Pull request merged**